fuel-relayer/src/pending_blocks.rs

```diff
@@ -0,0 +1,617 @@
+        let back_height = self.pending_block_commits.back().unwrap().block_height;
+
+        if height < back_height {
+            // This case means that we somehow skipped block commit and didnt receive it in expected order.
+            error!(
+                "Commited block {height} is lower then current lowest pending block {back_height}."
+            );
+        } else if height > front_height {
+            // check if we are lagging against da layer
+            if self.chain_height < height {
+                error!(
+                    "Our chain height: {} is lower then da layer height {height}",
+                    self.chain_height
+                );
+            }
+            // new block received. Happy path
+            if height == front_height + BlockHeight::from(1u64) {
+                self.pending_block_commits
+                    .push_front(PendingBlock::new_commited_block(
+                        da_height, height, block_root,
+                    ));
+            } else {
+                error!("Commited block height {height} should be only increased by one from current height {front_height}.");
+            }
+        } else {
+            // happens if reverted commit is again visible
+            // iterate over pending blocks and set reverted commit as commited.
+            for pending in self.pending_block_commits.iter_mut() {
+                if pending.block_height == height {
+                    if !pending.reverted {
+                        error!("We received block {height} commit that was not reverted")
+                    }
+                    pending.da_height = da_height;
+                    pending.reverted = false;
+                    pending.block_height = height;
+                    pending.block_root = block_root;
+                    break;
+                }
+            }
+        }
+    }
+
+    /// Handle revert of block commit from contract.
+    fn handle_block_commit_revert(
+        &mut self,
+        block_root: Bytes32,
+        height: BlockHeight,
+        da_height: DaBlockHeight,
+    ) {
+        if self.pending_block_commits.is_empty() {
+            // nothing to revert
+            error!("Revert for height {height} received while pending block queue is empty and LFCFB is {}",
+                self.last_commited_finalized_fuel_height);
+            return;
+        }
+
+        let front_height = self.pending_block_commits.front().unwrap().block_height;
+        let back_height = self.pending_block_commits.back().unwrap().block_height;
+
+        if height < back_height {
+            // ignore block that are not inside pending queue.
+            error!("All pending block commits should be present in block queue. height:{height} last_known:{back_height}");
+        } else if height > front_height {
+            error!("Something unexpected happened.Reverted block commits are not something found in the future.");
```

Comment:
Related to my other point, could this occur fairly regularly if the L1 contract doesn't emit any logs for a few blocks?